trio/_core/_tests/test_run.py

```diff
@@ -208,26 +227,26 @@ async def crasher():
             await sleep_forever()
 
 
-async def test_reschedule():
-    t1 = None
-    t2 = None
+async def test_reschedule() -> None:
+    t1: _core.Task | None = None
+    t2: _core.Task | None = None
 
-    async def child1():
+    async def child1() -> None:
         nonlocal t1, t2
         t1 = _core.current_task()
         print("child1 start")
         x = await sleep_forever()
         print("child1 woke")
         assert x == 0
         print("child1 rescheduling t2")
-        _core.reschedule(t2, outcome.Error(ValueError()))
+        _core.reschedule(not_none(t2), outcome.Error(ValueError()))
```

Comment:
Why not put the `assert` above this call? (similar for the one below)

Just a nitpick though, really.

Reply:
I added `not_none()` to try and distinguish between asserts used to test Trio, and ones used to indicate what should be happening here. If `t2` never gets set, there's either something wrong with the test or I guess Trio's core internals.